src/web/app/components/teammates-common/submission-status-tooltip.pipe.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { FeedbackSessionSubmissionStatus } from '../../../types/api-output';
+import { SubmissionStatusTooltipPipe } from './submission-status-tooltip.pipe';
+
+describe('SubmissionStatusTooltipPipe', () => {
+  it('create an instance', () => {
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transform with no deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasNoDeadlines = {
+      studentDeadlines: {},
+      instructorDeadlines: {},
+    };
+    const hasNoOngoingDeadlines = {
+      studentDeadlines: { nonOngoingExtension1: new Date('2019-01-01').valueOf() },
+      instructorDeadlines: { nonOngoingExtension2: new Date('2019-02-01').valueOf() },
+    };
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+
+    expect(pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN, hasNoDeadlines)).toEqual(
+      pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN),
+    );
+
+    expect(pipe.transform(FeedbackSessionSubmissionStatus.OPEN, hasNoOngoingDeadlines)).toEqual(
+      pipe.transform(FeedbackSessionSubmissionStatus.OPEN),
+    );
+  });
+
+  it('transform with deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasOngoingDeadlines = {
+      studentDeadlines: { ongoingDeadline: new Date('2021-01-01').valueOf() },
+      instructorDeadlines: { nonOngoingDeadline: new Date('2019-01-01').valueOf() },
+    };
+
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    const extensionMessage = 'with current ongoing session extensions.';
+
+    const notVisibileWithExtension = pipe.transform(FeedbackSessionSubmissionStatus.NOT_VISIBLE, hasOngoingDeadlines);
+    expect(notVisibileWithExtension.substring(
+      notVisibileWithExtension.length - extensionMessage.length, notVisibileWithExtension.length),
+    ).toEqual(extensionMessage);
```

Comment:
Feeling fancy today :)

```suggestion
    const expectedExtensionMessage = expect.stringMatching(/.+current ongoing session extensions\.$/);
    expect(notVisibileWithExtension).toEqual(expectedExtensionMessage);
```

Alternatively, can assert `notVisibleWithExtension.endsWith(extensionMessage)` is true. I think both are easier to read than the current substring solution.